opm/polymer/PolymerInflow.cpp

```diff
@@ -73,13 +73,13 @@ namespace Opm
     {
         ScheduleConstPtr schedule = eclipseState->getSchedule();
         for (const auto& well : schedule->getWells(currentStep)) {
-            if (well->isInjector(currentStep)) {
+            if (well->isInjector(currentStep) && well->getStatus(currentStep) != WellCommon::SHUT) {
                 WellInjectionProperties injection = well->getInjectionProperties(currentStep);
                 if (injection.injectorType == WellInjector::WATER) {
                     WellPolymerProperties polymer = well->getPolymerProperties(currentStep);
                     wellPolymerRate_.insert(std::make_pair(well->name(), polymer.m_polymerConcentration));
                 } else {
-                    OPM_THROW(std::logic_error, "For polymer injector you must have a water injector");
```

Comment:
>Well - polymer injection in a gas (or Oil!) injector is a fundamental error isn't it?

It will not happen, because we only look for water injector in the ``if`` statement.